sanic/app.py

```diff
@@ -896,7 +896,17 @@ async def handle_request(self, request: Request):  # no cov
                     )
 
                 # Run response handler
+                await self.dispatch(
+                    "http.handler.before",
+                    inline=True,
+                    context={"request": request},
+                )
                 response = handler(request, **request.match_info)
+                await self.dispatch(
+                    "http.handler.after",
+                    inline=True,
+                    context={"request": request},
+                )
```

Comment:
Wouldn't this be better after awaiting the handler (in those cases in which it is awaitable)?

Reply:
Yes. Good catch.